src/actions/waitforexternaleventaction.ts

```diff
@@ -0,0 +1,14 @@
+import { ActionType, IAction } from "../classes";
+
+/** @hidden */
+export class WaitForExternalEventAction implements IAction {
+    public readonly actionType: ActionType = ActionType.WaitForExternalEvent;
+
+    constructor(
+        public readonly externalEventName: string,
+    ) {
+        if (!externalEventName || typeof externalEventName !== "string") {
```

Comment:
For my own curiosity since we seem to do this same thing in many different places: why do we need to check the type of `externalEventName` if we've already declared it as a `string` a few lines above?

Reply:
TypeScript's types are a compile-time only thing and no type-checking is done at runtime (see a later comment below), and `externalEventName` is a user-supplied piece of information. If this check was removed, it would be possible for a user to write something like

`yield context.df.waitForExternalEvent({ name: "aBadIdea", willWork: false });`

in their orchestrator, which could lead to the orchestration shim or extension schema-parsing failing and throwing a less informative error further down the stack, or the `waitForExternalEvent` method looking for a history event whose event name matched the anonymous object in question, never finding it, and causing the orchestrator to hang without an easily visible cause.

Short version: TypeScript's types are for the person writing the TypeScript, not the code running against the transpiled JS.